Sources/performance-worker/BenchmarkClient.swift

```diff
@@ -80,18 +89,123 @@ struct BenchmarkClient {
+    switch result {
+    case .success:
+      errorCode = nil
+    case let .failure(error):
+      errorCode = error.code
+    }
+    return (
+      latency: Double(endTime - startTime), errorCode: errorCode
+    )
+  }
+
   // The result is the number of nanoseconds for processing the RPC.
   private func makeRPC(
-    client: Grpc_Testing_BenchmarkServiceClient,
+    benchmarkClient: Grpc_Testing_BenchmarkServiceClient,
     rpcType: Grpc_Testing_RpcType
-  ) -> (latency: Double, errorCode: RPCError.Code?) {
+  ) async throws -> (latency: Double, errorCode: RPCError.Code?) {
     switch rpcType {
-    case .unary, .streaming, .streamingFromClient, .streamingFromServer, .streamingBothWays,
-      .UNRECOGNIZED:
-      let startTime = DispatchTime.now().uptimeNanoseconds
-      let endTime = DispatchTime.now().uptimeNanoseconds
+    case .unary:
+      return try await self.computeTimeAndErrorCode { request in
+        let responseStatus = try await benchmarkClient.unaryCall(
+          request: ClientRequest.Single(message: request)
+        ) {
+          response in
+          return response.accepted
+        }
+
+        return responseStatus
+      }
+
+    // Repeated sequence of one request followed by one response.
+    // It is a ping-pong of messages between the client and the server.
+    case .streaming:
+      return try await self.computeTimeAndErrorCode { request in
+        let ids = AsyncStream.makeStream(of: Int.self)
+        let streamingRequest = ClientRequest.Stream { writer in
+          for try await id in ids.stream {
+            if id <= self.messagesPerStream {
+              try await writer.write(request)
+            } else {
+              return
+            }
+          }
+        }
+
+        ids.continuation.yield(1)
+
+        let responseStatus = try await benchmarkClient.streamingCall(request: streamingRequest) {
+          response in
+          var id = 1
+          for try await _ in response.messages {
+            id += 1
+            ids.continuation.yield(id)
+          }
+          return response.accepted
+        }
+
+        return responseStatus
+      }
+
+    case .streamingFromClient:
+      return try await self.computeTimeAndErrorCode { request in
+        let streamingRequest = ClientRequest.Stream { writer in
+          for _ in 1 ... self.messagesPerStream {
+            try await writer.write(request)
+          }
+        }
+
+        let responseStatus = try await benchmarkClient.streamingFromClient(
+          request: streamingRequest
+        ) { response in
+          return response.accepted
+        }
+
+        return responseStatus
+      }
+
+    case .streamingFromServer:
+      return try await self.computeTimeAndErrorCode { request in
+        let responseStatus = try await benchmarkClient.streamingFromServer(
+          request: ClientRequest.Single(message: request)
+        ) { response in
+          return response.accepted
+        }
```

Comment:
We need to consume the messages on the response: if we don't consume them we'll exert backpressure and the server will stop writing messages.